### PR TITLE
Table Batch / Entity Group Transactions v1

### PR DESCRIPTION
**Pull request by edwin-huber**
Unit Tests and Integration Tests added.

ToDos:
```
src\table\batch\BatchSerialization.ts
```
 - Validate behvaiour for GET HTTP verb / Query op
 - Improve RegEx
```
src\table\batch\TableBatchManager.ts
```
 - Implement all APIs used by batch / entity group transactions
```
General
```
 - Validate behaviour matches service in failure cases.
 - Validate header behaviors and requirements
 - Include tests from @zzhxiaofeng 
 - Correct handling of Content ID as per OData specification.

### Development Guideline

Please go to CONTRIBUTION.md for steps about setting up development environment and recommended Visual Studio Code extensions.

### CODE REVIEW

**Comment by edwin-huber** [comment]
Seems we have issues with node 8x and the new code... need to check this by installing an old version of node.

**Comment by edwin-huber** [comment]
There is still quite a bit of refactoring coming to improve the serialization code.  
This is dependent on the completion of test coverage and query handling.  
We are still working on :
- ensuring unit tests for serialization are correctly checking all response types.
- covering all query types

File: package.json [hunk 1]
```diff
@@ -67,6 +68,7 @@
     "mocha": "^5.2.0",
     "prettier": "^2.1.2",
     "prettier-tslint": "^0.4.2",
+    "to-readable-stream": "^2.1.0",
```
Comment (XiaoningLiu) [inline package.json:71]: Only need to add this dependency into "dependencies". Don't need in "devDependencies".

File: src/table/handlers/TableHandler.ts [hunk 2]
```diff
@@ -245,6 +249,8 @@ export default class TableHandler extends BaseHandler implements ITableHandler {
   }
 
   // TODO: Create data structures to hold entity properties and support serialize, merge, deserialize, filter
+  // Note: Batch is using the partition key and row key args, handler receives these values from middleware via
+  // context
```
Comment (XiaoningLiu) [inline src/table/handlers/TableHandler.ts:253]: minor: Change _partitionKey to partitionKey if we actually use it. Same with other variables.

File: src/table/batch/TableBatchOrchestrator.ts [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+   * see Link below for details of response format
+   * tslint:disable-next-line: max-line-length
+   * https://docs.microsoft.com/en-us/rest/api/storageservices/performing-entity-group-transactions#json-versions-2013-08-15-and-later-2
+   *
+   * @private
+   * @return {*}  {string}
+   * @memberof TableBatchManager
+   */
+  private serializeResponses(): string {
+    let responseString: string = "";
```
Comment (XiaoningLiu) [inline src/table/batch/TableBatchOrchestrator.ts:104]: If batch suboperation includes query, then one and only one query suboperation is allowed in batch.
We can check if Logic App/Durable Function demo requires this. Should be not.

File: src/table/handlers/TableHandler.ts (hunk 2 above)
Comment (edwin-huber) [reply src/table/handlers/TableHandler.ts:253]: done, removed from all handler functions

File: src/table/batch/TableBatchOrchestrator.ts (hunk 3 above)
Comment (edwin-huber) [reply src/table/batch/TableBatchOrchestrator.ts:104]: Adding as a work item

**Comment by edwin-huber** [comment]
Discussed and approved with @XiaoningLiu

**Pull request merged**